api/grpcserver/v2alpha1/layer_test.go

```diff
@@ -0,0 +1,225 @@
+package v2alpha1
+
+import (
+	"context"
+	"errors"
+	"io"
+	"testing"
+
+	spacemeshv2alpha1 "github.com/spacemeshos/api/release/go/spacemesh/v2alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/events"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/sql/blocks"
+	"github.com/spacemeshos/go-spacemesh/sql/layers"
+)
+
+func TestLayerService_List(t *testing.T) {
+	db := sql.InMemory()
+	ctx := context.Background()
+
+	lrs := make([]layers.Layer, 100)
+	for i := range lrs {
+		l, err := generateLayer(db, types.LayerID(i), true)
+		require.NoError(t, err)
+		lrs[i] = *l
+	}
+
+	svc := NewLayerService(db)
+	cfg, cleanup := launchServer(t, svc)
+	t.Cleanup(cleanup)
+
+	conn := dialGrpc(ctx, t, cfg)
+	client := spacemeshv2alpha1.NewLayerServiceClient(conn)
+
+	t.Run("limit set too high", func(t *testing.T) {
+		_, err := client.List(ctx, &spacemeshv2alpha1.LayerRequest{Limit: 200})
+		require.Error(t, err)
+
+		s, ok := status.FromError(err)
+		require.True(t, ok)
+		assert.Equal(t, codes.InvalidArgument, s.Code())
+		require.Equal(t, "limit is capped at 100", s.Message())
+	})
+
+	t.Run("no limit set", func(t *testing.T) {
+		_, err := client.List(ctx, &spacemeshv2alpha1.LayerRequest{})
+		require.Error(t, err)
+
+		s, ok := status.FromError(err)
+		require.True(t, ok)
+		assert.Equal(t, codes.InvalidArgument, s.Code())
+		require.Equal(t, "limit must be set to <= 100", s.Message())
+	})
+
+	t.Run("limit and offset", func(t *testing.T) {
+		list, err := client.List(ctx, &spacemeshv2alpha1.LayerRequest{
+			Limit:  25,
+			Offset: 50,
+		})
+		require.NoError(t, err)
+		require.Len(t, list.Layers, 25)
+	})
+
+	t.Run("all", func(t *testing.T) {
+		ls, err := client.List(ctx, &spacemeshv2alpha1.LayerRequest{
+			StartLayer: 0,
+			EndLayer:   100,
+			Limit:      100,
+		})
+		require.NoError(t, err)
+		require.Len(t, lrs, len(ls.Layers))
+	})
+}
+
+func TestLayerStreamService_Stream(t *testing.T) {
+	db := sql.InMemory()
+	ctx := context.Background()
+
+	lrs := make([]layers.Layer, 100)
+	for i := range lrs {
+		l, err := generateLayer(db, types.LayerID(i), true)
+		require.NoError(t, err)
+		lrs[i] = *l
+	}
+
+	svc := NewLayerStreamService(db)
+	cfg, cleanup := launchServer(t, svc)
+	t.Cleanup(cleanup)
+
+	conn := dialGrpc(ctx, t, cfg)
+	client := spacemeshv2alpha1.NewLayerStreamServiceClient(conn)
+
+	t.Run("all", func(t *testing.T) {
+		events.InitializeReporter()
+		t.Cleanup(events.CloseEventReporter)
+
+		stream, err := client.Stream(ctx, &spacemeshv2alpha1.LayerStreamRequest{})
+		require.NoError(t, err)
+
+		var i int
+		for {
+			_, err := stream.Recv()
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			i++
+		}
+		require.Len(t, lrs, i)
```

Comment:
How about asserting that it gets each expected layer (instead of, for example, each time the same "something")?